<a href="https://colab.research.google.com/github/Vynogradova/homework/blob/main/Homework%20%22%D0%A2%D0%B5%D1%81%D1%82%D0%B8_%D0%B4%D0%BB%D1%8F_%D0%BC%D0%B0%D0%BB%D0%B8%D1%85_%D0%B2%D0%B8%D0%B1%D1%96%D1%80%D0%BE%D0%BA_%D1%82%D0%B0_%D0%BF%D1%80%D0%BE%D0%BF%D0%BE%D1%80%D1%86%D1%96%D0%B9_ipynb%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Тести для малих вибірок та пропорцій



In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


1.
H₀: μ = 123.7  (нульова гіпотеза - середній час до покупки не зріс)  
H₁: μ > 123.7  (альтернативна гіпотеза - середній час до покупки зріс)  
Тип тесту: правосторонній (one-tailed)  


In [ ]:
2.
import numpy as np
from scipy import stats

time_to_purchase = np.array([128,135,121,142,126,151,123])
mu0 = 123.7

x_bar = np.mean(time_to_purchase)
s = np.std(time_to_purchase, ddof=1)
n = len(time_to_purchase)

print(f"Вибіркове середнє x_bar = {x_bar:.2f}")
print(f"Вибііркове стандартне відхилення s = {s:.2f}")
print(f"Вибірка n = {n}")

Вибіркове середнє x_bar = 132.29
Вибііркове стандартне відхилення s = 10.98
Вибірка n = 7


In [ ]:
3.
alpha = 0.10

#Крок 1. Обчислимо t-статистику вручну
t_stat = (x_bar - mu0) / (s / np.sqrt(n))

# Крок 2. Критичне значення для правостороннього тесту
df = n - 1
t_crit = stats.t.ppf( 1- alpha, df)

# Крок 3. p-value
p_value = 1 - stats.t.cdf(t_stat, df)

print(f"t-статистика: {t_stat:.3f}")
print(f"Критичне значення (df={df}): {t_crit:.3f}")
print(f"p-value: {p_value:.4f}")

# Висновок
if t_stat > t_crit:
    print(f"Відхиляємо H0: середній час до покупки зріс (рівень значущості {alpha})")
else:
    print("Не відхиляємо H0: доказів недостатньо")

t-статистика: 2.069
Критичне значення (df=6): 1.440
p-value: 0.0420
Не відхиляємо H0: доказів недостатньо


4.
Використовуючи p-value, ми відхиляємо H₀ на рівні α=0.10.

Значить, є статистично значуще збільшення часу до покупки після редизайну.

5.
Редизайн сайту, ймовірно, подовжив середній час до покупки для користувачів.

In [ ]:
alpha = 0.05

#Крок 1. Обчислимо t-статистику вручну
t_stat = (x_bar - mu0) / (s / np.sqrt(n))

# Крок 2. Критичне значення для правостороннього тесту
df = n - 1
t_crit = stats.t.ppf(1 - alpha, df)

# Крок 3. p-value
p_value = 1 - stats.t.cdf(t_stat, df)

print(f"t-статистика: {t_stat:.3f}")
print(f"Критичне значення (df={df}): {t_crit:.3f}")
print(f"p-value: {p_value:.4f}")

# Висновок
if t_stat > t_crit:
    print(f"Відхиляємо H0: середній час до покупки зріс (рівень значущості {alpha})")
else:
    print("Не відхиляємо H0: доказів недостатньо")

t-статистика: 2.069
Критичне значення (df=6): 1.943
p-value: 0.0420
Не відхиляємо H0: доказів недостатньо


6.
p-value ≈ 0.0420 < 0.05 → на рівні α=0.05 також відхиляємо H₀.

Висновок не змінюється: збільшення часу до покупки підтверджується.

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

In [28]:
import statsmodels.api as sm
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

count =  311  # кількість "успіхів"
nobs = 1200   # обсяг вибірки
p0 = 0.23      # гіпотетична пропорція (H0)
alpha = 0.03

# z-тест для пропорції
stat, pval = proportions_ztest(count, nobs, p0, alternative='larger')
confint = proportion_confint(count, nobs, alpha=0.03, method='normal')

print(" Тест для однієї пропорції")
print(f"Z-статистика = {stat:.3f}")
print(f"p-value = {pval:.4f}")
print(f"97% довірчий інтервал: {confint}")

 Тест для однієї пропорції
Z-статистика = 2.306
p-value = 0.0106
97% довірчий інтервал: (0.23171700302179205, 0.28661633031154127)


p-value < α (0.0106 < 0.03) - відхиляємо нульову гіпотезу H₀: p = 0.23

Це означає, що є статистично значущі підстави стверджувати, що частка дорослих, які впізнають логотип, зросла після рекламної кампанії.